src/components/settings/Users/EditUser.vue

```diff
@@ -0,0 +1,219 @@
+ */
+function Debounce() {
+    let timeout = null;
+
+    /**
+     * exec callback function after delay if no new call to function happens
+     * @param {function} callback function to execute after delay
+     * @param {number} [delay=100] delay before execute the callback if no new call to function happens
+     * @returns {void}
+     */
+    return function (callback, delay = 100) {
+        clearTimeout(timeout);
+        timeout = setTimeout(() => callback(), delay);
+    };
+}
+
+export default {
+    props: {
+        id: {
+            type: String,
+            required: true
+        }
+    },
+
+    data: () => ({
+        loading: false,
+        username: "",
+        saving: false,
+        currentPassword: "",
+        newPassword: "",
+        repeatNewPassword: "",
+        savingPassword: false,
+        active: false
+    }),
+
+    mounted() {
+        this.getUser();
+    },
+
+    methods: {
+        // Used to prevent dbl "click" evt fired by checkbox label
+        debounce: new Debounce(),
+
+        /**
+         * Get user from server
+         * @returns {void}
+         */
+        getUser() {
+            this.loading = true;
+            this.$root.getSocket().emit("getUser", this.id, (res) => {
+                this.loading = false;
+                if (res.ok) {
+                    const { username, active } = res.user;
+
+                    this.username = username;
+                    this.active = active;
+                } else {
+                    toast.error(res.msg);
+                }
+            });
+        },
+
+        /**
+         * Check new passwords match before saving it
+         * @returns {void}
+         */
+        savePassword() {
+            this.savingPassword = true;
+            const { currentPassword, newPassword, repeatNewPassword } = this;
+
+            if (newPassword !== repeatNewPassword) {
+                toast.error(this.$t("PasswordsDoNotMatch"));
+                this.savingPassword = false;
+                return;
+            }
+
+            this.$root
+                .getSocket()
+                .emit(
+                    "changePassword",
+                    this.id,
+                    {
+                        currentPassword,
+                        newPassword
+                    },
+                    (res) => {
+                        this.savingPassword = false;
+                        this.$root.toastRes(res);
+
+                        if (res.ok) {
+                            this.currentPassword = "";
+                            this.newPassword = "";
+                            this.repeatNewPassword = "";
+                        }
+                    });
+
+        },
+
+        /**
+         * Save user changes
+         * @param {Object} user - user to save
+         * @param {string} user.username - username used as login identifier.
+         * @param {boolean} user.active - is the user authorized to login?
```

Comment:
as used above these fileds may not be present 
```suggestion
         * @param {string|undefined} user.username - username used as login identifier.
         * @param {boolean|undefined} user.active - is the user authorized to login?
```

Reply:
You are right, I forgot to mark them as optional in the doc. However, your code suggestion seems wrong. [Following JSDoc](https://jsdoc.app/tags-param.html#optional-parameters-and-default-values), I have to change it to:
```js
        /**
         * Save user changes
         * @param {Object} user - user to save
         * @param {string} [user.username] - username used as login identifier.
         * @param {boolean} [user.active] - is the user authorized to login?
         * @returns {void}
         */
```
(Notice bracket `[]` around `user.username` and `user.active`).